common/patches/0001-Build-only-what-we-use-to-test-EBBR-on-32b-arm.patch

```diff
@@ -0,0 +1,1587 @@
+-	acpi/pptt/pptt.c 			\
+-	acpi/rasf/rasf.c			\
+-	acpi/rsdp/rsdp.c			\
+-	sbbr/rsdp/rsdp.c			\
+-	acpi/rsdt/rsdt.c			\
+-	acpi/s3/s3.c 				\
+-	acpi/s3power/s3power.c 			\
+-	acpi/s4/s4.c 				\
+-	acpi/sbst/sbst.c			\
+-	acpi/sdei/sdei.c			\
+-	acpi/sdev/sdev.c			\
+-	acpi/slic/slic.c 			\
+-	acpi/slit/slit.c 			\
+-	acpi/spcr/spcr.c 			\
+-	acpi/spmi/spmi.c 			\
+-	acpi/srat/srat.c 			\
+-	acpi/stao/stao.c			\
+-	acpi/syntaxcheck/syntaxcheck.c 		\
+-	acpi/tcpa/tcpa.c 			\
+-	acpi/tpm2/tpm2.c 			\
+-	acpi/uefi/uefi.c			\
+-	acpi/uniqueid/uniqueid.c		\
+-	acpi/waet/waet.c			\
+-	acpi/wakealarm/wakealarm.c 		\
+-	acpi/wdat/wdat.c			\
+-	acpi/wmi/wmi.c 				\
+-	acpi/wpbt/wpbt.c			\
+-	acpi/wsmt/wsmt.c			\
+-	acpi/xsdt/xsdt.c			\
+-	acpi/xenv/xenv.c			\
+-	apic/apicedge/apicedge.c 		\
+-	bios/bios_info/bios_info.c 		\
+-	bios/bios32/bios32.c 			\
+-	bios/ebda_region/ebda_region.c 		\
+-	bios/ebdadump/ebdadump.c 		\
+-	bios/hdaaudio/hdaaudio.c 		\
+-	bios/interrupt/interrupt.c		\
+-	bios/memmapdump/memmapdump.c 		\
+-	bios/mtrr/mtrr.c 			\
+-	bios/multiproc/mpcheck.c 		\
+-	bios/multiproc/mpdump.c 		\
+-	bios/pciirq/pciirq.c 			\
+-	bios/pnp/pnp.c 				\
+-	bios/romdump/romdump.c 			\
+-	bios/s0idle/s0idle.c 			\
+-	bios/smm/smm.c				\
+-	cmos/cmosdump/cmosdump.c 		\
+-	coreboot/clog/clog.c			\
+-	cpu/virt/virt.c 			\
+-	cpu/virt/virt_svm.c 			\
+-	cpu/virt/virt_vmx.c			\
+-	cpu/maxfreq/maxfreq.c 			\
+-	cpu/cpufreq/cpufreq.c 			\
+-	cpu/nx/nx.c 				\
+-	cpu/msr/msr.c 				\
+-	cpu/microcode/microcode.c 		\
+-	dmi/dmicheck/dmicheck.c 		\
+-	hotkey/hotkey/hotkey.c 			\
+-	ipmi/bmc/bmc_info.c			\
+ 	kernel/klog/klog.c 			\
+ 	kernel/olog/olog.c			\
+ 	kernel/oops/oops.c 			\
+ 	kernel/version/version.c 		\
+-	opal/mtd_info.c				\
+-	opal/prd_info.c				\
+-	pci/aspm/aspm.c 			\
+-	pci/crs/crs.c 				\
+-	pci/maxreadreq/maxreadreq.c 		\
+-	pci/smccc/smccc.c			\
+-	tpm/tpmevlog/tpmevlog.c			\
+-	tpm/tpmevlogdump/tpmevlogdump.c		\
+ 	uefi/csm/csm.c 				\
+ 	uefi/uefidump/uefidump.c 		\
+ 	uefi/uefirttime/uefirttime.c		\
+@@ -201,22 +42,14 @@ fwts_SOURCES = main.c 				\
+ 	uefi/uefirtauthvar/uefirtauthvar.c	\
+ 	uefi/esrtdump/esrtdump.c		\
+ 	uefi/esrt/esrt.c			\
+-	$(pci_tests)				\
+-	$(mem_tests)				\
+-	$(cpu_tests)				\
+-	$(power_mgmt_tests)			\
+ 	$(dt_tests)
+ 
+-fwts_LDFLAGS = -lm -lbsd `pkg-config --libs glib-2.0 gio-2.0`
++fwts_LDFLAGS = -lm
+ 
+ fwts_LDADD = \
+ 	-lfwts					\
+ 	-L$(top_builddir)/src			\
+-	-L$(top_builddir)/src/acpica		\
+-	-L$(top_builddir)/src/libfwtsiasl	\
+-	-L$(top_builddir)/src/libfwtsacpica	\
+-	-L$(top_builddir)/src/lib/src		\
+-	-lfwtsacpica
++	-L$(top_builddir)/src/lib/src
+ 
+ man_MANS = ../doc/fwts.1 ../doc/fwts-collect.1 ../doc/fwts-frontend-text.1
+ 
+diff --git a/src/lib/include/fwts.h b/src/lib/include/fwts.h
+index f34d4859..33486b30 100644
+--- a/src/lib/include/fwts.h
++++ b/src/lib/include/fwts.h
+@@ -127,6 +127,12 @@
+ #define FWTS_USE_DEVMEM 1
+ #endif
+ 
++#if defined(__arm__)
++#define FWTS_ARCH_ARM	1
```

Comment:
Query: Where are these defines finally used?

Reply:
Hi @edhay,
- `FWTS_HAS_UEFI` is used by the sources under `src/uefi/`.
- `FWTS_USE_DEVMEM` is used by `src/lib/src/fwts_mmap.c`.
- `FWTS_ARCH_ARM` should be used by sources like `src/lib/src/fwts_arch.c`, but I did not add the proper modifications there yet.